docs/source/user_guides/basic_conceptions.rst

```diff
@@ -0,0 +1,303 @@
+Basic Concepts
+--------------
+
+Introduction
+~~~~~~~~~~~~
+
+Dialog Flow Framework helps its users create conversational services, which is done by
+defining a specialized dialog graph that dictates the behaviour of the dialog service.
+This dialog graph essentially represents the dialog script that guides the conversation
+between the chat-bot and the user.
+
+The DFF leverages a specialized language known as a Domain-Specific Language (DSL)
+to enable developers to quickly write and comprehend dialog graphs.
+This DSL greatly simplifies the process of designing complex conversations and handling
+various user inputs, making it easier to build sophisticated conversational systems.
+
+DFF installation and requirements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For this very basic tutorial we will need only the core dependencies of DFF.
+They can be installed via the following command:
+
+.. code-block:: shell
+
+    pip3 install dff
+
+Example conversational chat-bot
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let us go through the creation of a simple bot that would play (virtual) ping-pong game with its users.
+It would also greet them and handle exceptions.
+First, we define the chat-bot in pseudo language:
+
+.. code-block:: text
+
+    If user writes "Hello!":
+        Respond with "Hi! Let's play ping-pong!"
+    
+    If user afterwards writes "Ping" or "ping" or "Ping!":
+        Respond with "Pong!"
+        Repeat this behaviour
+    
+    If user afterwards writes something else:
+        Respond with "That was against the rules!"
+        Repeat from the beginning if user writes anything
```

Comment:
We should change our bot definition, because right now all the advanced example adds that is related to this definition is support for regexp.

(and also the definition should be exactly the same as what the advanced example does)